src/cpu/aarch64/matmul/acl_lowp_matmul_sq.hpp

```diff
@@ -0,0 +1,326 @@
+                        bia_d.dims()[1], bia_d.dims()[0]));
+            }
+
+            arm_compute::GEMMLowpOutputStageInfo info;
+
+            info.type = arm_compute::GEMMLowpOutputStageType::
+                    QUANTIZE_DOWN_FIXEDPOINT;
+            info.gemmlowp_multiplier = 1073741824;
+            info.gemmlowp_shift = -1;
+            info.gemmlowp_offset = 0;
+            info.gemmlowp_min_bound = -128;
+            info.gemmlowp_max_bound = 127;
+            info.output_data_type = almc_.dst_tensor_info.data_type();
+
+            almc_.gemm_info.set_gemmlowp_output_stage(info);
+
+            auto scratchpad = scratchpad_registry().registrar();
+            const dnnl::impl::memory_desc_t dst_md_ {desc_.dst_desc};
+            arm_compute::ActivationLayerInfo act_info;
+            CHECK(init_scratchpad(engine, scratchpad, acl_post_ops,
+                    attr_.post_ops_, act_info, dst_md_));
+
+            almc_.gemm_info.set_activation_info(act_info);
+            ACL_CHECK_VALID(arm_compute::NEGEMMLowpMatrixMultiplyCore::validate(
+                    &almc_.src_tensor_info, &almc_.wei_tensor_info,
+                    almc_.with_bias ? &almc_.bia_tensor_info : nullptr,
+                    &almc_.dst_tensor_info, almc_.gemm_info));
+
+            return status::success;
+        }
+
+        status_t init_scratchpad(engine_t *engine,
+                memory_tracking::registrar_t &scratchpad,
+                acl_post_ops_t &post_ops, dnnl::impl::post_ops_t &attr_post_ops,
+                arm_compute::ActivationLayerInfo &act_info,
+                const dnnl::impl::memory_desc_t &dst_md) {
+            CHECK(post_ops.init(engine, attr_post_ops, dst_md, act_info));
+            // ACL only accepts s32 bias for quantization and since
+            // the current bias vector is f32 we need to convert.
+            if (almc_.with_bias) {
+                const memory_desc_wrapper bias_d(&bias_md_);
+                // printf("*** init_scratchpad: bias_data_type=%d\n", bias_md_.data_type);
+                scratchpad.book(
+                        memory_tracking::names::key_conv_bias_s32_convert,
+                        bias_d.nelems(), bias_d.data_type_size());
+            }
+            return status::success;
+        }
+
+        acl_lowp_matmul_sq_conf_t almc_;
+        acl_post_ops_t acl_post_ops;
+    };
+
+    acl_lowp_matmul_sq_t(const pd_t *apd) : primitive_t(apd) {}
+
+    status_t create_resource(
+            engine_t *engine, resource_mapper_t &mapper) const {
+
+        if (mapper.has_resource(this)) return status::success;
+
+        auto r = utils::make_unique<acl_lowp_matmul_sq_resource_t>();
+        if (!r) return status::out_of_memory;
+
+        CHECK(r->configure(pd()->almc_));
+
+        mapper.add(this, std::move(r));
+
+        return status::success;
+    }
+
+    status_t execute(const exec_ctx_t &ctx) const {
+        std::lock_guard<std::mutex> _lock {this->mtx};
```

Comment:
Is the mutex protection only about the quantization parameters that are reset, or is there something else due to constant weights?